src/server/qgshttprequesthandler.cpp

```diff
@@ -505,6 +511,8 @@ bool QgsHttpRequestHandler::startGetFeatureResponse( QByteArray* ba, const QStri
     format = "text/xml";
 
   setInfoFormat( format );
+  startStreamingResponse();
```

Comment:
@rldhont Thanks for fixing this issue!

To explain my take on the code `sendHeaders()` was also my first take on the issue, but I thought it would be nice to make it more explicit when you want to start a streaming response as opposed to a standard response. The `mStreamingResponse` could also be used to protect against programming errors in the form of multiple calls to `sendResponse()` outside a streaming response.